flutter/packages/duck_router/test/src/duck_router_test.dart

```diff
@@ -409,6 +409,73 @@ void main() {
 
       expect(result, equals(1));
     });
+
+    testWidgets('Child back button dispatcher handles back button press',
+        (tester) async {
+      final config = DuckRouterConfiguration(
+        initialLocation: RootLocation(),
+      );
+
+      final router = await createRouter(config, tester);
+
+      // Navigate to a child route
+      router.navigate(to: HomeLocation());
+      await tester.pumpAndSettle();
+
+      expect(find.byType(HomeScreen), findsOneWidget);
+
+      // Simulate a back button press
+      await tester.binding.handlePopRoute();
+      await tester.pumpAndSettle();
+
+      // Check if we're still on the HomeScreen (we shouldn't be)
+      expect(find.byType(HomeScreen), findsNothing);
+
+      // We should have returned to the RootLocation, but we haven't
```

Comment:
I find this comment + assertion confusing. Where should we be, where are we?

Reply:
Let me know if it's clearer now